### PR TITLE
remove most arrow function and .bind lint errors

### PR DESCRIPTION
**Pull request by dkent600**
This is in an effort to minimize spurious react component rendering by removing most of the uses of arrow and .bind functions  in `render`. 

Down to only 11 lint warnings.

But I don't honestly see any significant performance improvements.

### CODE REVIEW

File: src/components/Proposal/Voting/VoteButtons.tsx [hunk 1]
```diff
@@ -91,20 +89,20 @@ class VoteButtons extends React.Component<IProps, IState> {
...
           "Voting requires reputation in " + dao.name :
-          proposal.stage === IProposalStage.ExpiredInQueue || (proposal.stage === IProposalStage.Boosted && expired) || (proposal.stage === IProposalStage.QuietEndingPeriod && expired)  || (proposal.stage === IProposalStage.Queued && expired) ?
+          proposal.stage === IProposalStage.ExpiredInQueue ||
+              (proposal.stage === IProposalStage.Boosted && expired) ||
+              (proposal.stage === IProposalStage.QuietEndingPeriod && expired)  ||
+              (proposal.stage === IProposalStage.Queued && expired) ?
             "Can't vote on expired proposals" :
             proposal.stage === IProposalStage.Executed ?
-              "Can't vote on executed proposals" :
-              `Vote ${vote === IProposalOutcome.Pass ? "for" : "against"}`
```
Comment (tibetsprague) [inline src/components/Proposal/Voting/VoteButtons.tsx]: im confused, why did you get rid of this one?

File: src/layouts/AppContainer.tsx [hunk 2]
```diff
@@ -163,31 +184,9 @@ class AppContainer extends React.Component<IProps, IState> {
...
-                    timestamp={timestamp}
-                    dismiss={() => dismissNotification(id)}
-                    showNotification={showNotification}
-                    minimize={() => this.setState({notificationsMinimized: true})}
-                  />
-                </div>
-              ))
+              <MinimizedNotifications notifications={sortedNotifications.length} unminimize={this.unminimizeNotif()} />
+              :
+              sortedNotifications.map(this.sortedNotification())
```
Comment (tibetsprague) [inline src/layouts/AppContainer.tsx]: this looks weird, maybe name the function something different?

File: src/components/Account/AccountProfilePage.tsx [hunk 3]
```diff
@@ -177,7 +178,8 @@ class AccountProfilePage extends React.Component<IProps, null> {
 
                 return errors;
               }}
-              onSubmit={this.handleSubmit.bind(this)}
+              onSubmit={this.handleFormSubmit()}
```
Comment (jellegerbrandy) [inline src/components/Account/AccountProfilePage.tsx]: This still creates a new function object on each render, and so I'm not sure what is the improvement here.

The pattern to use here is to ` bind` the function int he constructor, as we do for example here: https://github.com/daostack/alchemy/blob/dev/src/components/Proposal/Create/SchemeForms/CreateContributionRewardProposal.tsx#L97

(So you would undo the change you made in `handleFormSubmit`, remove the ` bind` in onSubmit, and instead bind int he constructor)

File: src/components/Account/AccountProfilePage.tsx [hunk 4]
```diff
@@ -258,8 +260,8 @@ class AccountProfilePage extends React.Component<IProps, null> {
                         }
 
                         <h3>Social Verification</h3>
-                        <OAuthLogin editing={editing} provider="twitter" accountAddress={accountAddress} onSuccess={this.onOAuthSuccess.bind(this)} profile={accountProfile} socket={socket} />
-                        <OAuthLogin editing={editing} provider="github" accountAddress={accountAddress} onSuccess={this.onOAuthSuccess.bind(this)} profile={accountProfile} socket={socket} />
+                        <OAuthLogin editing={editing} provider="twitter" accountAddress={accountAddress} onSuccess={this.onOAuthSuccess()} profile={accountProfile} socket={socket} />
```
Comment (jellegerbrandy) [inline src/components/Account/AccountProfilePage.tsx]: see comment above about binding

File: src/components/Account/AccountProfilePage.tsx [hunk 5]
```diff
@@ -258,8 +260,8 @@ class AccountProfilePage extends React.Component<IProps, null> {
...
+                        <OAuthLogin editing={editing} provider="github" accountAddress={accountAddress} onSuccess={this.onOAuthSuccess()} profile={accountProfile} socket={socket} />
```
Comment (jellegerbrandy) [inline src/components/Account/AccountProfilePage.tsx]: see comment above about binding

File: src/components/Dao/DaoContainer.tsx [hunk 6]
```diff
@@ -80,31 +93,25 @@ class DaoContainer extends React.Component<IProps, null> {
           </div>
           <Switch>
             <Route exact path="/dao/:daoAvatarAddress/history"
-              render={(props) => <DaoHistoryPage {...props} currentAccountAddress={currentAccountAddress} />} />
+              render={this.daoHistoryRoute()} />
```
Comment (jellegerbrandy) [inline src/components/Dao/DaoContainer.tsx]: see comment above about binding

File: src/components/Dao/DaoContainer.tsx [hunk 7]
```diff
@@ -80,31 +93,25 @@ class DaoContainer extends React.Component<IProps, null> {
...
             <Route exact path="/dao/:daoAvatarAddress/members"
-              render={(props) => <DaoMembersPage {...props} daoState={daoState} />} />
+              render={this.daoMembersRoute()} />
```
Comment (jellegerbrandy) [inline src/components/Dao/DaoContainer.tsx]: see comment above about binding

File: src/components/Proposal/Voting/VoteButtons.tsx (hunk 1 above)
Comment (dkent600) [reply src/components/Proposal/Voting/VoteButtons.tsx]: That was a really weird piece of code.  Note that `tipContent` was a function before, taking the parameter `vote`.  Where was `vote` supposed to come from, and what did it represent?  I could find no answers so I removed it.

File: src/layouts/AppContainer.tsx (hunk 2 above)
Comment (dkent600) [reply src/layouts/AppContainer.tsx]: done

File: src/components/Account/AccountProfilePage.tsx (hunk 3 above)
Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: @tibetsprague You once advised me to use the method I'm using.  Can you please confirm @jellegerbrandy's suggestion, to make sure we're all on the same page?  Thanks!

(Just a note BTW, documenting best practices for fixing lint errors like this one is one of the documentation tasks on your plate, Tibet  :-)  )

Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: @jellegerbrandy   @tibetsprague  My understanding of the problem here is not that the function is being called on every render, but worse, the fact the react thinks a new function is created actually causes any subcomponents depending on it (to which the new function is given) to be rendered more often.

Reading some more:  There is a similar pattern with the creation of new instances of _anything_, like variables created in the render method, and that are then passed into subcomponents.

File: src/components/Proposal/Voting/VoteButtons.tsx (hunk 1 above)
Comment (tibetsprague) [reply src/components/Proposal/Voting/VoteButtons.tsx]: looks pretty obvious in the code, you pass in whether it is a pass button or fail button you are mousing over and then the tip says Vote For or Vote Against. im not sure when that funcionality was removed but it used to exist. dont know if we care, no one has mentioned it.

Comment (dkent600) [reply src/components/Proposal/Voting/VoteButtons.tsx]: this text is only being used when voting is disabled, so the case you describe can't happen.

File: src/components/Account/AccountProfilePage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: The code is correct, even if it is not optimal. `this.handleFormSubmit()` returns the actual handler function, bound to this.  @tibetsprague this is the pattern you long ago recommended to me.  :-)

File: src/components/Account/AccountProfilePage.tsx (hunk 4 above)
Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: no, `this.onOAuthSuccess` is correct, if not optimal.

File: src/components/Account/OAuthLogin.tsx [hunk 8]
```diff
@@ -76,7 +76,7 @@ export default class OAuthLogin extends React.Component<IProps, IState> {
   // Kicks off the processes of opening the popup on the server and listening
   // to the popup. It also disables the login button so the user can not
   // attempt to login to the provider twice.
-  public startAuth(e: any) {
+  public startAuth = () => (e: any) => {
```
[the remark this answers is not in the document]
Comment (dkent600) [reply src/components/Account/OAuthLogin.tsx]: yes, I think as long as it is an arrow function it should be automatically bound to `this` and therefore give us what we want.

File: src/components/Account/AccountProfilePage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: @tibetsprague In any case, this is a big "doh" on my part....should've realized this.  My bad.

**Comment by dkent600** [comment]
@tibetsprague   @jellegerbrandy I believe I've corrected all the handler function functions.

File: src/components/Account/AccountProfilePage.tsx (hunk 3 above)
Comment (jellegerbrandy) [reply src/components/Account/AccountProfilePage.tsx]: well, yes, indeed.

[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply src/components/Account/AccountProfilePage.tsx]: So what is the conclusion here? 

Methinks we should still replace `handleFormSubmit()`  with `handleformsubmit` (and change the definition of the function and remove the first `() =>` from it.

[the remark this answers is not in the document]
Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: > Methinks we should still replace handleFormSubmit() with handleformsubmit (and change the definition of the function and remove the first () => from it.

That is what I've done  :-)

[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply src/components/Account/AccountProfilePage.tsx]: "yes, indeed", meaning that passing `this.handleFormSubmit()` will create a new function. 

We want to just pass `handleFormSubmit` to the subcomponent, and either:

-  keep the original handleFormSubmit, and ` bind()`  it in the constructor of the component.
- or define `public hanldeFormSubmit = (values) => { doit }` (so you do not need to `bind()` the function)

or am I misunderstanding something here?

[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply src/components/Account/AccountProfilePage.tsx]: exactly like you did below in `handleCloseMenu`, to me it looks like you  forgot this one.

**Comment by dkent600** [comment]
@jellegerbrandy 

> exactly like you did below in handleCloseMenu, to me it looks like you forgot this one

Indeed, my global search missed a case.  I've corrected them

File: src/components/Account/AccountProfilePage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkent600) [reply src/components/Account/AccountProfilePage.tsx]: Indeed, my global search missed a case. I've corrected them

**Pull request merged**